std-bits/base/src/main/java/org/enso/base/Encoding_Utils.java

```diff
@@ -165,12 +166,48 @@ private static ReportingStreamDecoder create_stream_decoder(InputStream stream,
   /**
    * A helper function which runs an action with a created stream decoder and closes it afterwards.
    */
-  public static <R> R with_stream_decoder(
+  public static <R> Pair<List<String>, R> with_stream_decoder(
```

Comment:
Should we use another `WithProblems<T>` here?
I know the pair is doing the same but feels cleaner.

Reply:
I guess you are right. I avoided using it because:
- the original `WithProblems` is in another package (this will probably be more and more common, at some point we may need to figure a way to be able to share Java code between our "native" extensions),
- it has a slightly different structure - there we had different possible classes of problems, each with a separate class; here we just need to store strings, because all problems will be converted into `Encoding_Error` at Enso side. Still I think a named class for this may be worth it, so I can add one.

Perhaps a
```java
public record WithProblems<ResultType, ProblemType>(ResultType result, List<ProblemType> problems) {}
```

Reply:
Pushed a changeset.